jolla-settings/pages/shaketorch/mainpage.qml

```diff
@@ -0,0 +1,102 @@
+import QtQuick 2.1
+import Sailfish.Silica 1.0
+import Nemo.DBus 2.0
+import Nemo.Configuration 1.0
+
+Page {
+    id: page
+
+    //% "ShakeTorch"
+    //: name in the Settings Page
+    property string entryname: qsTrId("settings_shaketorch_entryname")
+
+    // bind to Service
+    property bool serviceActive: (dbus.activeState == "active")
+    property bool serviceEnabled: (dbus.unitFileState == "enabled")
+
+    function refresh() {
+      onoffSwitch.checked = !onoffSwitch.checked
+      onoffSwitch.busy = false
+    }
+
+    /*
+     * Dbus interface to systemd manager
+     */
+    DBusInterface {
+        id: manager
+        bus: DBus.SessionBus
+        service: 'org.freedesktop.systemd1'
+        path: "/org/freedesktop/systemd1"
+        iface: 'org.freedesktop.systemd1.Manager'
+
+        signalsEnabled: true
+
+        function jobRemoved( id, job, unit, result ) {
+            if ( (unit === "harbour-shaketorch.service") && (result == "done") ) {
+                page.refresh()
+            }
+        }
+    }
+    /*
+     * Dbus interface to systemd unit
+     */
+    DBusInterface {
+        id: dbus
+        bus: DBus.SessionBus
+        service: "org.freedesktop.systemd1"
+        path: "/org/freedesktop/systemd1/unit/harbour_2dshaketorch_2eservice"
+        iface: "org.freedesktop.systemd1.Unit"
+
+        propertiesEnabled: true
+        signalsEnabled: true
+
+        property string activeState
+        property string subState
+        property string unitFileState
+
+        function startUnit() { call("Start", "replace", undefined, undefined ) }
+        function stopUnit()  { call("Stop",  "replace", undefined, undefined ) }
```

Comment:
Should we disable the service as well? So that if I stop it, it doesn't start at the next boot?

Reply:
The new text is much much clear now. Still, I don't expect the service to start at the next boot.

In my mind, if I want to disable the functionality I want to disable it until I go in the Settings and I re-enable it, not just until the next reboot.

I don't see any problem in stopping && disabling the service or starting && enabling the service in the same action.

Reply:
No worries, I can wait :-)

If you don't have time I can try to add this behavior myself. You did a **great** job already :pray: